schematic/schemas/data_model_parser.py

```diff
@@ -0,0 +1,319 @@
+            model_dict = jsonld_parser.parse_jsonld_model(self.path_to_data_model)
+        else:
+            raise ValueError(f"Schematic only accepts models of type CSV or JSONLD, you provided a model type {self.model_type}, please resubmit in the proper format.")
+
+        return model_dict
+    
+class DataModelCSVParser():
+    def __init__(
+        self
+        ):
+        # Instantiate DataModelRelationships
+        self.dmr = DataModelRelationships()
+        # Load relationships dictionary.
+        self.rel_dict = self.dmr.define_data_model_relationships()
+        # Get edge relationships
+        self.edge_relationships_dictionary = self.dmr.define_edge_relationships()
+        # Load required csv headers
+        self.required_headers = self.dmr.define_required_csv_headers()
+
+    def check_schema_definition(self, model_df: pd.DataFrame) -> bool:
+        """Checks if a schema definition data frame contains the right required headers.
+        Args:
+            schema_definition: a pandas dataframe containing schema definition; see example here: https://docs.google.com/spreadsheets/d/1J2brhqO4kpeHIkNytzlqrdIiRanXDr6KD2hqjOTC9hs/edit#gid=0
+        Raises: Exception
+        """
+        try:
+            if set(self.required_headers).issubset(set(list(model_df.columns))):
+                return
+            elif "Requires" in list(model_df.columns) or "Requires Component" in list(
+                model_df.columns
+            ):
+                raise ValueError(
+                    "The input CSV schema file contains the 'Requires' and/or the 'Requires "
+                    "Component' column headers. These columns were renamed to 'DependsOn' and "
+                    "'DependsOn Component', respectively. Switch to the new column names."
+                )
+            logger.debug("Schema definition csv ready for processing!")
+        except:
+            raise ValueError(
+                f"Schema extension headers: {set(list(model_df.columns))} "
+                f"do not match required schema headers: {self.required_headers}"
+            )
+        return
+
+
+    def gather_csv_attributes_relationships(self, model_df: pd.DataFrame) -> Dict:
+        '''Parse csv into a attributes:relationshps dictionary to be used in downstream efforts.
+        Args:
+            model_df: pd.DataFrame, data model that has been loaded into pandas DataFrame.
+        Returns:
+            attr_rel_dictionary: dict, 
+                {Attribute Display Name: {
+                    Relationships: {
+                                    CSV Header: Value}}}
+        '''
+        # Check csv schema follows expectations.
+        self.check_schema_definition(model_df)
+
+        # Get the type for each value that needs to be submitted.
+        # using csv_headers as keys to match required_headers/relationship_types
+        self.rel_val_types = {v['csv_header']:v['type']for k, v in self.rel_dict.items() if 'type' in v.keys()}
```

Comment:
If you're going to set an attribute in a method, it should probably be set in the `__init__` method as well, even if it just as `None`

Reply:
I'll move it up to init